pkg/tenancy/interface.go

```diff
@@ -0,0 +1,31 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package tenancy
+
+import (
+	"fmt"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+const regexOR = "|"
+
+var ErrUnauthorizedTenant = fmt.Errorf("authorized or invalid tenant")
+
+// ReadAuthorizer tells if a read request is allowed to query via Promscale.
+type ReadAuthorizer interface {
+	// ApplyTenantMatcher applies a safety matcher to incoming query matchers. This safety matcher is responsible
+	// from prevent unauthorized query reads from tenants that the incoming query is not supposed to read.
+	ApplyTenantMatcher(ms []*labels.Matcher) []*labels.Matcher
```

Comment:
Nit: while not a deal breaker, its obvious even from the function signature that we are appending the tenant matcher hence the function should be called `AppendTenantMatcher`.